testdata/p4_16_samples/bitvec_hdr.p4

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2018 MNK Consulting, LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <v1model.p4>
+
+struct alt_t {
+  bit<1> valid;
+  bit<7> port;
+};
+
+struct row_t {
+  alt_t alt0;
+  alt_t alt1;
+};
+
+header bitvec_hdr {
+  row_t row;
+}
+
+struct col_t {
+  bitvec_hdr bvh;
+}
+
+struct local_metadata_t {
+  row_t row0;
+  row_t row1;
+  col_t col;
+  bitvec_hdr bvh0;
+  bitvec_hdr bvh1;
+};
+
+struct parsed_packet_t {
+    bitvec_hdr bvh0;
+    bitvec_hdr bvh1;   
+};
+
+struct tst_t {
+  row_t row0;
+  row_t row1;
+  col_t col;  
+  bitvec_hdr bvh0;
+  bitvec_hdr bvh1;
+}
+
+parser parse(packet_in pk, out parsed_packet_t h,
+             inout local_metadata_t local_metadata,
+             inout standard_metadata_t standard_metadata) {
+  state start {
+    pk.extract(h.bvh0);
+    pk.extract(h.bvh1);
+    transition accept;
+  }
+}
+
+control ingress(inout parsed_packet_t h,
+                inout local_metadata_t local_metadata,
+	        inout standard_metadata_t standard_metadata) {
+
```

Comment:
Please add also a local variable with a type that is a nested header, and use it, for example, by copying the data from a different header and using its fields as the key. We have to check that the compiler handles using such types in other contexts as well.